### ISSUE TITLE
GetViewFrame fails with invalid frame when calling too much

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#802]
I use  FakeScreen to get a screenshot of what will be rendered in current frame without GUI and mouse cursors so I can apply pixel effects on top.

When doing it for a long time I get hit by an error saying the frame is invalid when I get a character current frame. In my game, loops don't have the same number of frames.

I figured I can use the function below to prevent this from happening. Note I need to check if the current frame number is valid - it should always be.

```
ViewFrame * getCharacterViewFrame(Character * chara) {
    int i_view = chara.View;
    int i_loop = chara.Loop;
    int i_frame = chara.Frame;
    if(i_loop >= Game.GetLoopCountForView(i_view)){
        i_loop =0;
    }
    if(i_frame >= Game.GetFrameCountForLoop(i_view, i_loop)){
        i_frame =0;
    }
    return Game.GetViewFrame(i_view, i_loop, i_frame);
}
```

**Comment by ivan-mogilko** [comment]
The error is supposedly not in GetViewFrame itself but in Character.View, Chracter.Loop and Character.Frame properties that return inconsistent values under certain condition.

**Comment by AlanDrake** [comment]
ChangeView mishaps during repex or late_repex?
Is it possible to make a reliable test that triggers the issue?

**Comment by ericoporto** [comment]
I cannot reproduce this anymore, so I am closing this issue.

**Comment by ivan-mogilko** [comment]
So, one possible case found by @AlanDrake is that when character is turning and not all directional loops contain full list of frames. Because engine does not reset the frame number in that case, then Character.Frame will report a frame which is not present in the current loop.

**Comment by ericoporto** [comment]
So, in my original problem, I had different frame count for each loop (8 frames for left and right, 6 for down, and 5 for up), and I would occasionally, but repeatedly hit this issue - I use the GetViewFrame for shadows and reflections in my game. 

But I couldn't trigger this in a small test game so I don't know what code was causing this to happen. My game source changed a lot and I ended up not hitting this - at the time I put the protection mentioned above, but eventually I removed it and it doesn't happen nowadays. :/
